ash: fix for bb bug #1087 - segfault on command "echo $(</tmp/file)"

 http://git.busybox.net/busybox/commit/?id=7426920461738367205e3d08649c827c747bd93d

diff -uBp busybox-1.15.3/shell/ash.c busybox/shell/ash.c
--- busybox-1.15.3/shell/ash.c	2010-02-20 03:00:00.000000000 +0300
+++ busybox/shell/ash.c	2010-02-20 11:26:15.000000000 +0300
@@ -4554,7 +4554,7 @@ forkchild(struct job *jp, union node *n,
 	if (mode == FORK_NOJOB          /* is it `xxx` ? */
 	 && n && n->type == NCMD        /* is it single cmd? */
 	/* && n->ncmd.args->type == NARG - always true? */
-	 && strcmp(n->ncmd.args->narg.text, "trap") == 0
+	 && n->ncmd.args && strcmp(n->ncmd.args->narg.text, "trap") == 0
 	 && n->ncmd.args->narg.next == NULL /* "trap" with no arguments */
 	/* && n->ncmd.args->narg.backquote == NULL - do we need to check this? */
 	) {
@@ -4642,7 +4642,7 @@ forkchild(struct job *jp, union node *n,
 	}
 #if JOBS
 	if (n && n->type == NCMD
-	 && strcmp(n->ncmd.args->narg.text, "jobs") == 0
+	 && n->ncmd.args && strcmp(n->ncmd.args->narg.text, "jobs") == 0
 	) {
 		TRACE(("Job hack\n"));
 		/* "jobs": we do not want to clear job list for it,
